docs/adr/0012-next-two-factor-authentication.md

```diff
@@ -0,0 +1,71 @@
+# Next Two Factor authentication in FxA
+
+- Status: accepted
+- Deciders: Vijay Budhram
+- Date: 2020-04-01
+
+## Context and Problem Statement
+
+Firefox Account originally implemented multi-factor authentication (MFA) support in Q1 of 2018.
+This feature used TOTP based codes and was based on RFC 6238.
+Additionally, if users lost their MFA device, they could use one time recovery codes to regain access to the account.
+
+Having MFA support has helped secure our users' accounts and given them more security flexibility.
+However, over time it has become more obvious that users that lose their MFA device (usually phone) are at risk of getting locked out of their account because they don't save their recovery codes.
+There are a non-trivial amount of users that don’t save or download their recovery codes, which is currently the only way they can regain access.
+In 2019 Q4, FxA started requiring that users confirm recovery codes before enabling MFA.
+While this did help reduce lockouts, we still wanted to reduce it further.
+
+We believe that adding a new MFA method to Firefox Accounts that has the similar security properties as TOTP would allow users to have another method to recover their account.
+
+## Decision Drivers
+
+- Improve user's account security
+- Reduce the risk of account lockout because of lost device/recovery code
+- Could be completed in roughly a summer
```

Comment:
I was going to suggest we reword summer to a "quarter" (or two quarters if applicable) but is the "summer" terminology used because this is part of the intern project, spanning a summer?